jujugui/static/gui/src/app/state/state.js

```diff
@@ -449,7 +449,24 @@ const State = class State {
     }
     const search = this.current.search;
     if (search) {
-      path = path.concat([PATH_DELIMETERS.get('search'), search]);
+      path.push(PATH_DELIMETERS.get('search'));
+      // Append the text if it is truthy, i.e. not a blank string etc.
+      if (search.text) {
+        path.push(search.text);
+      }
+      // Everything that is not the 'text' param should be appened as a query
+      // string.
+      delete search.text;
```

Comment:
This is deleting the key from the actual state object which we don't want to do because this will effectively be modifying history. Instead It's best to check in the loop below and continue if the `key === 'text'`